aspnetcore/includes/managed-identities-test-non-production.md

```diff
@@ -0,0 +1,14 @@
+---
+author: wpickett
+ms.author: wpickett
+ms.date: 08/22/2024
+ms.topic: include
+---
+> [!NOTE]
+> In this tutorial, a local database is used that doesn't use a password. For production apps:
+> * A password is required.
+> * Microsoft recommends that you use the most secure authentication flow available.
+>
+> Azure SQL Database should use [Managed Identities for Azure resources](/sql/connect/ado-net/sql/azure-active-directory-authentication#using-managed-identity-authentication). For non-Azure apps, use a secure authentication flow similar to managed identities for Azure resources.
+>
+> When the app is deployed to a test server, an environment variable can be used to set the connection string to a test database server. For more information, see [Configuration](xref:fundamentals/configuration/index).
```

Comment:
Sorry I keep refining what I think we need. I don't think we need too much warning as this is an onboarding tutorial where almost everyone is using a local db.

```suggestion
> This tutorial uses a local database without a password. Production apps should use the most secure authentication flow available. For more information on authentication for deployed test and production apps , see [Secure authentication flows](/aspnet/core/security/<!-- write create article something like secureAuthn-->).
```
@tdykstra @wadepickett  what do you think?

Reply:
I appreciate the help and the thought you are putting into this, thanks.

Would this existing topic possibly work to point to?  It does a good job of helping to establish which scenario applies and what to look into from there for auth flows: 
https://learn.microsoft.com/en-us/entra/identity-platform/authentication-flows-app-scenarios

If not, would it at least work for now until a better topic is created?

Reply:
> I appreciate the help and the thought you are putting into this, thanks.
> 
> I think we are back to the last one Tom updated of yours for now, until we have a better topic written to point to and then we can update the include in one place. That sound good?

Great. That's the power of includes.

Reply:
OK, I will not commit your shorter version above then, which would be great to use once we have a new topic that goes  into those details instead.  Thanks.

Reply:
I would tweak the intro to the include a bit:

```suggestion
This article uses a local database that doesn't require the user to be authenticated. Production apps should use the most secure authentication flow available. For more information on authentication for deployed test and production apps, see [Secure authentication flows](/aspnet/core/security/<!-- write create article something like secureAuthn-->).
```

Reply:
Thanks @tdykstra . That is better and improves the last commit suggestion for a shorter one from Rick we are using sometime in the future when there is a new article that would include what is needed so we don't have to be wordy as above.  That would be for the future when we have that article.

It is this longer version (what is actually committed, example below) I think we are going with for now.  Is this below OK as is for now?

> In this tutorial, a local database is used that doesn't use a password. For production apps:
> * A password is required.
> * Microsoft recommends that you use the most secure authentication flow available.
>
> Azure SQL Database should use [Managed Identities for Azure resources](/sql/connect/ado-net/sql/azure-active-directory-authentication#using-managed-identity-authentication). For non-Azure apps, use a secure authentication flow similar to managed identities for Azure resources.
>
> When the app is deployed to a test server, an environment variable can be used to set the connection string to a test database server. For more information, see [Configuration](xref:fundamentals/configuration/index).

Reply:
If so and there is a review approval for using that last commited longer include for now I will merge.  Thanks again for all the help.